packages/next/client/route-loader.ts

```diff
@@ -164,13 +197,15 @@ function resolvePromiseWithTimeout<T>(
       resolve(r)
     }).catch(reject)
 
-    requestIdleCallback(() =>
-      setTimeout(() => {
-        if (!cancelled) {
-          reject(err)
-        }
-      }, ms)
-    )
+    onBuildCallback(() => {
```

Comment:
@ijjk How about this? I'm wondering if this should be integrated into `requestIdleCallback`...